### PR TITLE
Prac 09 feedback

### PR DESCRIPTION
**Pull request by talitha1234**
Hey @Lucas-obrien could you review my code please? 

In particular, whether I have properly used association with my band class and how I have chosen to return a string in the repr method. Is there a better way to do this?

### CODE REVIEW

File: prac_09/unreliable_car.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    """Unreliable car class"""
+
+    def __init__(self, name: str, fuel: float, reliability: float):
+        """Construct unreliable car"""
+        super().__init__(reliability)
+        self.name = name
+        self.fuel = fuel
+        self.reliability = reliability
+
+    def drive(self, distance):
```
Comment (Lucas-obrien) [inline prac_09/unreliable_car.py:21]: Consider calling the super().drive here, we should only need to add a check to see if it starts.

File: prac_09/taxi_simulator.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    """Taxi simulator with drive and choose taxi options"""
+    taxis = [Taxi("Prius", 100), SilverServiceTaxi("Limo", 100, 2), SilverServiceTaxi("Hummer", 200, 4)]
+    current_taxi = None
+    bill_to_date = 0
+    choice = input(MENU).upper()
+    while choice != 'Q':
+        if choice == 'C':
+            current_taxi = get_current_taxi(current_taxi, taxis)
+        elif choice == 'D':
+            fare = drive_current_taxi(current_taxi)
```
Comment (Lucas-obrien) [inline prac_09/taxi_simulator.py]: Consider adding this directly to your bill_to_date

File: prac_09/taxi_simulator.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    current_taxi.drive(distance)
+    print(f'Your {current_taxi.name} trip cost you ${current_taxi.get_fare():,.2f}')
+    return current_taxi.get_fare()
+
+
+def get_current_taxi(current_taxi, taxis):
+    """Get a valid taxi choice from list displayed"""
+    display_taxis(taxis)
+    is_valid_taxi_number = False
+    while not is_valid_taxi_number:
```
Comment (Lucas-obrien) [inline prac_09/taxi_simulator.py]: Great repetition work here but it is a little beyond the scope, just referring back to the sample from the prac09, on an invalid input the code should be exiting back to the menu

File: prac_09/band.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    def add(self, musician):
+        """Add musician to band"""
+        self.band_members.append(musician)
+
+    def play(self):
+        """Return a string showing the instrument playing their first (or no) instrument."""
+        players_playing = []
+        for musician in self.band_members:
+            try:
+                players_playing.append(f"{musician.name} is playing: {musician.instruments[0]}")
```
Comment (Lucas-obrien) [inline prac_09/band.py]: Consider calling the super() here for each musician, everything else here is great!

File: prac_09/silver_service_taxi.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
+"""
+Silver Service Taxi
+Estimate: 15 mins
+Actual: 20 mins
+"""
+
+from prac_09.taxi import Taxi
+
+
+class SilverServiceTaxi(Taxi):
```
Comment (Lucas-obrien) [inline prac_09/silver_service_taxi.py:10]: Just missing some docstrings here

File: prac_09/taxi_simulator.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+Estimate: 30 mins
+Actual: 70 mins
+"""
+from prac_09.silver_service_taxi import SilverServiceTaxi
+from prac_09.taxi import Taxi
+
+MENU = 'q)uit, c)hoose taxi, d)rive\n>>> '
+
+
+def main():
```
Comment (Lucas-obrien) [inline prac_09/taxi_simulator.py:12]: Just missing a docstring here

**Review by Lucas-obrien — COMMENTED** [review]
Hey Talitha, just a few minor things to look at. Code structures look great!

File: prac_09/unreliable_car.py (hunk 1 above)
Comment (talitha1234) [reply prac_09/unreliable_car.py:21]: thank you this makes a lot more sense now why we are inheriting the class

File: prac_09/taxi_simulator.py (hunk 2 above)
Comment (talitha1234) [reply prac_09/taxi_simulator.py]: good point, I don't need the extra line

File: prac_09/taxi_simulator.py (hunk 3 above)
Comment (talitha1234) [reply prac_09/taxi_simulator.py]: Thanks for pointing this out, I will change to suit the sample.

File: prac_09/band.py (hunk 4 above)
Comment (talitha1234) [reply prac_09/band.py]: Great note, will fix this up!

File: prac_09/silver_service_taxi.py (hunk 5 above)
Comment (talitha1234) [reply prac_09/silver_service_taxi.py:10]: Thanks for highlighting this.

File: prac_09/taxi_simulator.py (hunk 6 above)
Comment (talitha1234) [reply prac_09/taxi_simulator.py:12]: I appear to have a docstring there but I'll add the menu choices to make it more clear. Thanks!

Comment (talitha1234) [reply prac_09/taxi_simulator.py:12]: I appear to have a docstring but I will add the menu to make this more clear. Thanks!

**Pull request merged**